client/core/core.go

```diff
@@ -267,10 +267,6 @@ func (c *Core) tryCancel(dc *dexConnection, oid order.OrderID) (found bool, err
 		return
 	}
 
-	// Lock at least until the cancel is added to the trackedTrade struct.
-	dc.tradeMtx.Lock()
-	defer dc.tradeMtx.Unlock()
```

Comment:
Not needed, and was a bad attempt at synchronization here anyways. May have been a holdover from when the mutexes were all over the place.